argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,253 @@
+# <Copyright 2020, Argo AI, LLC. Released under the MIT license.>
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+import click
+from argoverse.data_loading.object_label_record import read_label
+from argoverse.evaluation.detection_utils import (
+    DistFnType,
+    SimFnType,
+    dist_fn,
+    filter_annos,
+    get_error_types,
+    get_ranks,
+    interp,
+    sim_fn,
+)
+from tqdm.contrib.concurrent import process_map
+
+
+@dataclass
+class DetectionCfg:
+    """Instantiates a DetectionCfg object for configuring a DetectionEvaluator.
+
+    Args:
+        sim_ths: The similarity thresholds for determining a true positive.
+        sim_fn_type: The type of similarity function to be used for calculating average precision.
+        n_rec_samples: The number of recall points to sample uniformly in [0, 1].
+        wdcs: The weight vector for the weighted detection composite score (WDCS).
+        tp_threshold: The center distance threshold for the true positive metrics.
+        significant_digits: The precision for metrics.
+    """
+
+    sim_ths: Tuple[float] = (0.5, 1.0, 2.0, 4.0)
+    sim_fn_type: SimFnType = SimFnType.CENTER
+    n_rec_samples: int = 101
+    ads_weighting: Tuple[float] = (3, 1, 1, 1)
+    tp_thresh: float = 2.0
+    significant_digits: int = 3
+
+
+@dataclass
+class DetectionEvaluator:
```

Comment:
might be nice to add some `logging` output through the execution, so the user can now what's going on

Reply:
Done!

Reply:
Thanks!